Userland/Libraries/LibWeb/DOM/Text.cpp

```diff
@@ -117,4 +117,33 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Text>> Text::split_text(size_t offset)
     return new_node;
 }
 
+// https://dom.spec.whatwg.org/#dom-text-wholetext
+String Text::whole_text() 
+{
+    // https://dom.spec.whatwg.org/#contiguous-text-nodes
+    // The contiguous Text nodes of a node node are node, node’s previous sibling Text node, if any, and its contiguous 
+    // Text nodes, and node’s next sibling Text node, if any, and its contiguous Text nodes, avoiding any duplicates. 
+    Vector<Text*> nodes;
+
+    nodes.prepend(this);
+
+    auto* current_node = previous_sibling();
+    while (current_node && current_node->is_text()) {
```

Comment:
I think this condition needs to be `while (current_node && (current_node->is_text() || current_node->is_cdata_section()))`, as `CDataSection` nodes should be included here.

This comment applies to the loop below as well.